### DISCUSSION TITLE
Jupyter notebooks version control

### DISCUSSION THREAD

**Discussion by cmichelenstrofer** [sandialabs/WecOptTool#121] [Dev]
Always clear the notebook before committing. Never commit output cells. 

Do this [automatically with a pre-commit hook](https://medium.com/somosfit/version-control-on-jupyter-notebooks-6b67a0cf12a3).

1. Install extension
```bash
pip install jupyter_contrib_nbextensions
```

2. Create file`.git/hooks/pre-commit`:
```bash
#!/bin/sh
jupyter nbconvert --ClearOutputPreprocessor.enabled=True --inplace examples/*.ipynb
git add .
```
This will look for all Jupyter notebooks in the `examples` folder.

3. Make file an executable
```bash
chmod +x .git/hooks/pre-commit
```

**Comment by cmichelenstrofer** [comment]
If there are no output cells, `nbsphinx` will run the notebooks when creating the documentation. So the tutorials ideally don't take too long to run.

**Comment by rgcoe** [comment]
For reference, here's some more explanation of how this works: https://medium.com/somosfit/version-control-on-jupyter-notebooks-6b67a0cf12a3
